src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -267,16 +267,73 @@ Copyright (c) .NET Foundation. All rights reserved.
     <AdditionalProbingPath Include="$(UserProfileRuntimeStorePath)$([System.IO.Path]::DirectorySeparatorChar)|arch|$([System.IO.Path]::DirectorySeparatorChar)|tfm|" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <CompileDependsOn>
+      $(CompileDependsOn)
+      _CreateAppHost;
+    </CompileDependsOn>
+  </PropertyGroup>
+
+  <!--
+    ============================================================
+                                        _CreateAppHost
+    If we found a restored apphost, create the modified destiniation apphost
+    with options from the project.
+    ============================================================
+     -->
+  <UsingTask TaskName="CreateAppHost" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateAppHost"
+          DependsOnTargets="_GetAppHostPaths;CoreCompile"
+          Condition="'$(ComputeNETCoreBuildOutputFiles)' == 'true'">
+    <PropertyGroup>
+      <UseWindowsGraphicalUserInterface Condition="$(RuntimeIdentifier.StartsWith('win')) and '$(OutputType)'=='WinExe'">true</UseWindowsGraphicalUserInterface>
```

Comment:
Does this need to be overrideable? Since it's set inside the target there's nothing the user can do to override it in their project, so perhaps it should have an underscore "private" name.

Reply:
This is a preexisting property from a previous commit, so I think we can make it "private" in a follow-up refactoring PR.

Reply:
Made it private.